modules/ide/xsk-ide-xsaccess/src/main/resources/META-INF/dirigible/xsk-ide-xsaccess/editor/editor.js

```diff
@@ -0,0 +1,169 @@
+      let xhr = new XMLHttpRequest();
+      xhr.open('GET', thePath, false);
+      xhr.send();
+      if (xhr.status === 200) {
+        return xhr.responseText;
+      }
+    }
+
+    function loadContents(xsaccessfile) {
+      if (!xsaccessfile) {
+        console.error('XSACCESS file parameter is not present in the URL');
+      } else {
+        return getResource('/services/v4/ide/workspaces' + xsaccessfile);
+      }
+    }
+
+    function getViewParameters() {
+      if (window.frameElement.hasAttribute("data-parameters")) {
+        let readParams = JSON.parse(window.frameElement.getAttribute("data-parameters"));
+        $scope.file = readParams["file"];
+      } else {
+        let scanSearchParams = new URLSearchParams(window.location.search);
+        $scope.file = scanSearchParams.get('file');
+      }
+    }
+
+    $scope.initJSONcontent = function () {
+      if (!('allowMethods' in $scope.data.cors))
+        $scope.data.cors = {
+          ...$scope.data.cors,
+          allowMethods: $scope.allAllowedMethods,
+          allowOrigin: ["*", "*.sap.com"],
+          maxAge: 3600,
+          allowHeaders: [''],
+          exposeHeaders: ['']
+        };
+      if ($scope.data.headers.customHeaders && $scope.data.headers.customHeaders.length)
+        $scope.data.headers.customHeaders = $scope.data.headers.customHeaders.map((x) => {
+          return {
+            ...x,
+            thevalue: x.value.substr(0, 10) == 'ALLOW-FROM' ? 'ALLOW-FROM' : x.value,
+            url: x.value.substr(0, 10) == 'ALLOW-FROM' ? x.value.substr(11) : ''
+          }
+        })
+    }
+
+    function load() {
+      getViewParameters();
+      contents = loadContents($scope.file);
+      $scope.data = JSON.parse(contents);
+      $scope.anonymous_connection_allowed = $scope.data.anonymous_connection ? true : false;
+      $scope.allAllowedMethods = ["GET", "POST", "HEAD", "OPTIONS"];
+      $scope.initJSONcontent();
+    }
+
+    load();
+
+    function updateObjectNestedProp(obj, value, propPath) {
+      const [head, ...rest] = propPath.split('.');
+      if (!rest.length) {
+        obj[head] = value;
+      } else {
+        updateObjectNestedProp(obj[head], value, rest.join('.'));
+      }
+    }
+
+    function getNestedValue(obj, path) {
+      return path.split('.').reduce(function (prev, curr) {
+        return prev[curr];
+      }, obj || this);
+    }
+
+    $scope.deleteInList = function (path, index) {
+      let newKeyValue = getNestedValue($scope.data, path).filter((_val, num) => num != index);
+      if (!newKeyValue.length) newKeyValue.push('');
+      updateObjectNestedProp(
+        $scope.data,
+        newKeyValue,
+        path);
+    }
+
+    $scope.addInList = function (path, value) {
+      let newKeyValue = getNestedValue($scope.data, path);
+      newKeyValue.push(value);
+      updateObjectNestedProp($scope.data,
+        newKeyValue,
+        path);
+    }
+
+    $scope.trigCorsMethod = function (method) {
```

Comment:
Do not shorten names. Use `triggerCorsMethod`

Reply:
OK